### ISSUE TITLE
figure out how to do oauth login with Flask/Python/etc

### ISSUE THREAD

**Issue by alexrudnick** [hltdi/guampa#1]
Write a wiki page about how to do it and/or do a demo.

We want to be able to log in with probably Google or Facebook.

**Comment by alexrudnick** [comment]
Consider [flask-oauthlib](https://github.com/lepture/flask-oauthlib) ?

**Comment by alexrudnick** [comment]
flask-oauthlib doesn't seem to have an example for how to log in with Google.

Can the flask-oauth examples be modified to work?
https://github.com/mitsuhiko/flask-oauth/tree/master/example

**Comment by alexrudnick** [comment]
This package looks like it does fb and twitter and google!

https://github.com/omab/python-social-auth

**Comment by alexrudnick** [comment]
Oy. Made that demo go on Python 3. I should maybe send them an update for the requirements, for Python 3...

https://github.com/omab/python-social-auth/blob/master/examples/flask_example/requirements.txt

**Comment by alexrudnick** [comment]
Aaaand it doesn't seem to work with OAuth, just OpenID. This means we can easily log in with Google and Yahoo! accounts, but not Twitter or Facebook.

**Comment by alexrudnick** [comment]
OK, after all of that, I'm going to go with [Mozilla's Persona](http://www.mozilla.org/en-US/persona/) login system, unless anybody has worries about it. But I've already implemented login and account creation with Persona, and it wan't all that hard :)

It was too much of a pain to make all of the Flask plugins for OAuth go, especially for Python 3. But Persona is really easy to implement, for a consumer -- I was able to adapt the example that _comes with Flask_ (and is like 60 lines of Python) in an afternoon, _and_ implement account creation on top of that, in about a day total.

Closing this for now -- but if we decide we really need a "login with facebook" button, somebody else can feel free to implement it!
